rasa/nlu/components.py

```diff
@@ -91,13 +91,40 @@ def validate_requirements(component_names: List[Optional[Text]]) -> None:
         )
 
 
+def validate_component_keys(
+    component: "Component", component_config: Dict[Text, Any]
+) -> None:
+    """Validates that all keys for a component are valid.
+
+    Raises:
+        UserWarning: If the component has a key specified that is not used
+            by the component class, it is likely a mistake in the pipeline
+
+    Args:
+        component: The component class
+
+        component_config: The user-provided config for the component in the pipeline
+    """
+    component_name = component_config.get("name")
+    allowed_keys = set(component.defaults.keys())
+    provided_keys = set(component_config.keys())
+    provided_keys.remove("name")
```

Comment:
optional: `pop` with a default might be more robust in case the `name` item doesn't exist for some reason

Reply:
(`remove` raises in case the element is not in the set)

Reply:
`pop` with a set is for arbitrary elements, could be a comprehension though
```suggestion
    provided_keys = (key for key in provided_keys if not key == "name")
```